test/extended/util/cvo.go

```diff
@@ -0,0 +1,57 @@
+package util
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclientv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+)
+
+// WaitForClusterProgression waits for a cluster-level configuration change to propagate across all operators.
+// This polls all ClusterVersion objects and waits until the following states are stable:
+//
+// 1. Available = true
+// 2. Progressing = false
+// 3. Failing = false
+//
+// If a ClusterVersion reports it is failing, this will abort with an error.
+func WaitForClusterProgression(cv configclientv1.ClusterVersionsGetter) error {
```

Comment:
how to do you ensure this doesn't return true *before* the change starts rolling out?

Reply:
thought of a "dumb" method - ensure the wait loop is successful 3 times before ending the polling. Otherwise we'd have to know all subscribers to whatever thing we are changing.